core/src/main/java/tc/oc/pgm/api/channels/Channel.java

```diff
@@ -0,0 +1,118 @@
+package tc.oc.pgm.api.channels;
+
+import static net.kyori.adventure.text.Component.text;
+
+import cloud.commandframework.arguments.standard.StringArgument;
+import cloud.commandframework.paper.PaperCommandManager;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.event.ChannelMessageEvent;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.util.Players;
+import tc.oc.pgm.util.named.NameStyle;
+
+public interface Channel<T> {
+
+  default String getDisplayName() {
+    return getAliases()[0];
+  }
+
+  String[] getAliases();
+
+  @Nullable
+  default Character getShortcut() {
+    return null;
+  }
+
+  default SettingValue getSetting() {
+    return null;
+  }
+
+  default boolean supportsRedirect() {
+    return false;
+  }
+
+  default boolean canSendMessage(MatchPlayer sender) {
+    return true;
+  }
+
+  T getTarget(MatchPlayer sender, Map<String, ?> arguments);
+
+  Collection<MatchPlayer> getViewers(T target);
+
+  default Collection<MatchPlayer> getBroadcastViewers(T target) {
+    return getViewers(target);
+  }
+
+  default void sendMessage(ChannelMessageEvent<T> event) {}
```

Comment:
```suggestion
  void sendMessage(ChannelMessageEvent<T> event);
```
either force the downstream to implement this, or have an *actual* default impl, but don't just leave an empty no-op method for something this critical.
I'd personally advocate for having a default impl here, just like broadcastMessage has